src/format/OpVaultReaderBandEntry.cpp

```diff
@@ -0,0 +1,324 @@
+            return nullptr;
+        }
+        entK_key = keyAndMacKey.mid(0, keyLen);
+        entK_hmacKey = keyAndMacKey.mid(keyLen, hmacLen);
+    }
+
+    QString dKeyB64 = bandEntry.value("d").toString();
+    OpData01* entD01 = OpData01::decode01Base64(dKeyB64, entK_key, entK_hmacKey);
+    if (entD01->hasError()) {
+        qCritical() << R"(Unable to decipher "d" in UUID ")" << uuid << "\": " << entD01->errorString();
+        delete entD01;
+        return nullptr;
+    }
+    const QByteArray entD = *(entD01->getClearText());
+    delete entD01;
+
+    const QJsonDocument dDoc = QJsonDocument::fromJson(entD);
+    QJsonObject obj = dDoc.object();
+
+#ifdef DUMP_OPVAULT_JSON
+    qDebug("entry d is <<%s>>\n", dDoc.toJson(QJsonDocument::Indented).data());
+#endif /*DUMP_OPVAULT_JSON*/
+
+    auto result = new QPair<QJsonObject, QPair<QByteArray, QByteArray>>();
+    result->first = obj;
+    result->second = *(new QPair<QByteArray, QByteArray>(entK_key, entK_hmacKey));
+    return result;
+}
+
+Entry* OpVaultReader::processBandEntry(const QJsonObject& bandEntry, const QDir& attachmentDir)
+{
+    const QString uuid = bandEntry.value("uuid").toString();
+    if (not(uuid.size() == 32 or uuid.size() == 36)) {
+        qWarning() << QString("Skipping suspicious band UUID <<%1>> with length %2").arg(uuid).arg(uuid.size());
+        return nullptr;
+    }
+
+    const auto entry = new Entry();
+
+    if (bandEntry.contains("category")) {
+        const QJsonValue& categoryValue = bandEntry["category"];
+        if (categoryValue.isString()) {
+            bool found = false;
+            const QString category = categoryValue.toString();
+            for (Group* group : this->m_db->rootGroup()->children()) {
+                const QVariant& groupCode = group->property("code");
+                if (category == groupCode.toString()) {
+                    entry->setGroup(group);
+                    found = true;
+                    break;
+                }
+            }
+            if (!found) {
+                qWarning() << QString("Unable to place Entry.Category \"%1\" so using the Root instead").arg(category);
+                entry->setGroup(this->m_db->rootGroup());
+            }
+        } else {
+            qWarning() << QString(R"(Skipping non-String Category type "%1" in UUID "%2")")
+                              .arg(categoryValue.type())
+                              .arg(uuid);
+            entry->setGroup(this->m_db->rootGroup());
+        }
+    } else {
+        qWarning() << "EGAD, category-less entry <<\n" << bandEntry << "\n>> in UUID " << uuid;
+        entry->setGroup(this->m_db->rootGroup());
+    }
+
+    entry->setUpdateTimeinfo(false);
+    {
+        TimeInfo ti;
+        bool timeInfoOk = false;
+        if (bandEntry.contains("created")) {
+            const qint64 createdTime(bandEntry["created"].toInt());
+            ti.setCreationTime(QDateTime::fromSecsSinceEpoch(createdTime, Qt::UTC));
```

Comment:
QDateTime::fromSecsSinceEpoch was [introduced in Qt 5.8](http://doc.qt.io/qt-5/qdatetime.html#fromSecsSinceEpoch); prior to that it was (I believe) QDateTime::fromTime_t. There are additional calls to this function in this PR.